controlplane/kubeadm/controllers/controller.go

```diff
@@ -271,16 +296,24 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	case numMachines < desiredReplicas && numMachines == 0:
 		// Create new Machine w/ init
 		logger.Info("Initializing control plane", "Desired", desiredReplicas, "Existing", numMachines)
+		conditions.MarkFalse(controlPlane.KCP, controlplanev1.ResizeSucceededCondition, controlplanev1.ScalingUpReason, clusterv1.ConditionSeverityWarning, "Scaling up to %d replicas (actual %d)", desiredReplicas, numMachines)
 		return r.initializeControlPlane(ctx, cluster, kcp, controlPlane)
 	// We are scaling up
 	case numMachines < desiredReplicas && numMachines > 0:
 		// Create a new Machine w/ join
 		logger.Info("Scaling up control plane", "Desired", desiredReplicas, "Existing", numMachines)
+		conditions.MarkFalse(controlPlane.KCP, controlplanev1.ResizeSucceededCondition, controlplanev1.ScalingUpReason, clusterv1.ConditionSeverityWarning, "Scaling up to %d replicas (actual %d)", desiredReplicas, numMachines)
 		return r.scaleUpControlPlane(ctx, cluster, kcp, controlPlane)
 	// We are scaling down
 	case numMachines > desiredReplicas:
 		logger.Info("Scaling down control plane", "Desired", desiredReplicas, "Existing", numMachines)
+		conditions.MarkFalse(controlPlane.KCP, controlplanev1.ResizeSucceededCondition, controlplanev1.ScalingDownReason, clusterv1.ConditionSeverityWarning, "Scaling down to %d replicas (actual %d)", desiredReplicas, numMachines)
 		return r.scaleDownControlPlane(ctx, cluster, kcp, controlPlane)
+	default:
+		// make sure last resize operation is marked as completed (if any)
+		if conditions.Has(controlPlane.KCP, controlplanev1.ResizeSucceededCondition) {
+			conditions.MarkTrue(controlPlane.KCP, controlplanev1.ResizeSucceededCondition)
+		}
```

Comment:
Similar to above around upgrades, this doesn't necessarily feel like it's marking a resizing operation as successful as much as indicating the lack of resizing needed currently. 

I'm wondering if it might be better to name this condition so that it reflects that better. `HasRequestedSize` is the best I'm currently able to come up with, but I'm not overly happy with it either.

Reply:
As agreed, we are removing from this PR the ResizeSucceededCondition